hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/HazelcastJsonValueCodec.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.codec.builtin;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.core.HazelcastJsonValue;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Codec for SQL JSON data type.
+ *
+ * @since 5.1
+ */
+public final class HazelcastJsonValueCodec {
```

Comment:
@sancar @ihsandemir What do you guys think about this codec? I feel like it should have been a custom codec (rather than a manually written one), but doing so will cause it to have 2 more extra frames (BEGIN_FRAME, STRING_FRAME, END_FRAME rather than just a STRING_FRAME)

Reply:
It looks like it is more of question of whether we will extend HazelcastJsonValue with new fields on the future or not. 

Some background context for @ivanthescientist 
BEGIN_FRAME and END_FRAME is to make sure that they are extensible without breaking old readers.  
To do this you are gonna need to use https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/custom/Custom.yaml to generate this codec automatically

Keeping in mind that it is a highly used feature and can be optimized/refactored in the future, I would vote for playing safe and making it a custom(auto-generated) codec.

Reply:
I do not see why JSON value would be extended in the future? Can you think of such a need? If no such a case in mind, i would rather not over-engineer, keep it to minimum bytes.

Reply:
@ihsandemir I don't have anything specific right now in my mind. I was just being cautious.

Reply:
These kind of over-engineering for the future usually is not worth it, as i happened to see such features we added in the past, which was never used. Hence, I would go with minimum bytes.